src/lib/bcmp/bm/bm_l2.cpp

```diff
@@ -354,8 +357,24 @@ err_t bm_l2_rx(void* device_handle, uint8_t* payload, uint16_t payload_len, uint
 err_t bm_l2_link_output(struct netif *netif, struct pbuf *pbuf) {
     (void) netif;
 
-    /* Send on all available ports (Multicast) */
-    return bm_l2_tx(pbuf, bm_l2_ctx.available_ports_mask);
+    // by default, send to all ports
+    uint8_t port_mask = bm_l2_ctx.available_ports_mask;
+
+    // if the application set an egress port, send only to that port
+    constexpr size_t bcmp_egress_port_offset_in_dest_addr = 13;
+    const size_t egress_port_idx = sizeof(struct eth_hdr) + offsetof(struct ip6_hdr, dest) +
+                                   bcmp_egress_port_offset_in_dest_addr;
+    uint8_t *eth_frame = static_cast<uint8_t *>(pbuf->payload);
+    uint8_t egress_port = eth_frame[egress_port_idx];
+    constexpr uint8_t num_adin_ports = 2; // TODO generalize
+    if (egress_port > 0 && egress_port <= num_adin_ports) {
+      port_mask = 1 << (egress_port - 1);
+    }
```

Comment:
Question for my own understanding, here we will set the `port_mask = 1 << (egress_port - 1)` , where egress_port will be 1 or 2 meaning `port_mask` will equal  0b01 or 0b10 and these correspond to the two different ports?

Reply:
You've got it exactly. The field that comes in via the IP address is a number (1 or 2), and the ADIN tx function needs a bit mask of which ports to send on, where port 1 is the LSB (0x01), port 2 is the next bit (0x10). If both bits are set (0x11, the default) then the ADIN transmits on both ports.